### PR TITLE
DOC-6146: Archive Service updates

### PR DESCRIPTION
**Pull request by AnnaK-R3**

### CODE REVIEW

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
+# Archive Service release notes
+
+The Archive Service is a standalone service that operates on a different release cadence to the Corda platform.
+
+The Archive Service compatibility with Corda Enterprise:
+
+| Archive Service version      | Corda Enterprise version    |
+|------------------------------|-----------------------------|
+| 1.0.x                        | 4.10 and below              |
+| 1.1.x                        | 4.11 and above              |
```
Comment (adelel1) [inline content/en/tools/archiving-service/archiving-release-notes.md:21]: And 4.11.x and above

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 2]
```diff
@@ -1,43 +1,73 @@
...
+If you deviate from the above compatibility guidelines, for example, using the Archive Service 1.1.x with Corda Enterprise 4.10, the Archive Service will not work.
+{{< /note >}}
+
+## Corda Enterprise 4.11 and above
+
+### Archive Service 1.1
+
+In this release:
+
+* Version 1.1 is a compatibility release of the Archive Service. This and all future 1.1.x releases will only work with Corda 4.11 and above and will not be compatible with Corda Enterprise 4.10.x and below. Use the latest Archive Service 1.0.x for Corda Enterprise 4.10.x and below.
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/tools/archiving-service/archiving-release-notes.md]: Updated

File: content/en/tools/archiving-service/archiving-release-notes.md (hunk 1 above)
Comment (AnnaK-R3) [reply content/en/tools/archiving-service/archiving-release-notes.md]: Updated

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 3]
```diff
@@ -1,43 +1,73 @@
...
+
+# Archive Service release notes
+
+The Archive Service is a standalone service that operates on a different release cadence to the Corda platform.
+
+The Archive Service compatibility with Corda Enterprise:
+
+| Archive Service version      | Corda Enterprise version    |
+|------------------------------|-----------------------------|
+| 1.0.x                        | 4.10 and below              |
```
[the remark this answers is not in the document]
Comment (AnnaK-R3) [reply content/en/tools/archiving-service/archiving-release-notes.md]: Updated

**Review by adelel1 — APPROVED** [review]

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 4]
```diff
@@ -1,43 +1,73 @@
...
+* Version 1.1 is a compatibility release of the Archive Service. This and all future 1.1.x releases will only work with Corda 4.11.x and above and will not be compatible with Corda Enterprise 4.10.x and below. Use the latest Archive Service 1.0.x for Corda Enterprise 4.10.x and below.
+* In Corda Enterprise 4.11 a new column has been added to the node transactions table for additional signatures. The new Archive Service release includes this new column in its snapshot data.
+
+## Corda Enterprise 4.10 and below
+
+### Archive Service 1.0.5
+
+In this release:
+
+* Previously, an issue occurred where the Archive Service misunderstood reference states. This would prevent a transaction from being archivable if it referenced an unconsumed transaction and did not have any unconsumed transactions that referenced it.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:42]:
```suggestion
#### Fixed issues
The Archive Service misunderstood reference states. As a result, the Archive Service could not archive a transaction if it referenced an unconsumed transaction and there were no unconsumed transactions that referenced it.
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 5]
```diff
@@ -1,43 +1,73 @@
...
+
+  Now, a transaction that only has outbound references (to unconsumed transactions) and no inbound references (from unconsumed transactions) will be marked as archivable by the Archive Service. Any inbound references will still make a transaction unarchivable.
+
+### Archive Service 1.0.4
+
+In this release:
+
+* Previously, when using a Microsoft SQL Server database, an error was generated when using the `vault-states` filter with a positive value specified for the `retentionDays` parameter; this issue has been resolved.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:50]:
```suggestion
#### Fixed issues
When using a Microsoft SQL Server database, an error was generated when using the `vault-states` filter with a positive value specified for the `retentionDays` parameter.
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 6]
```diff
@@ -1,43 +1,73 @@
...
+
+In this release:
+
+* Previously, an issue occurred where the Archive Service misunderstood reference states. This would prevent a transaction from being archivable if it referenced an unconsumed transaction and did not have any unconsumed transactions that referenced it.
+
+  Now, a transaction that only has outbound references (to unconsumed transactions) and no inbound references (from unconsumed transactions) will be marked as archivable by the Archive Service. Any inbound references will still make a transaction unarchivable.
+
+### Archive Service 1.0.4
+
+In this release:
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:48]:
```suggestion
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 7]
```diff
@@ -1,43 +1,73 @@
...
+
+* Previously, when using a Microsoft SQL Server database, an error was generated when using the `vault-states` filter with a positive value specified for the `retentionDays` parameter; this issue has been resolved.
+
+### Archive Service 1.0.3
+
+In this release:
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:54]:
```suggestion
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 8]
```diff
@@ -1,43 +1,73 @@
...
+
+* A new configuration option has been added which allows the Archive Service to skip transactions that have legacy contract states that cause exceptions during a JSON snapshot export. This configuration option is: `ignoreSnapshotExportFailures: true`.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:56]:
```suggestion
A new configuration option has been added which allows the Archive Service to skip transactions that have legacy contract states that cause exceptions during a JSON snapshot export. This configuration option is: `ignoreSnapshotExportFailures: true`.
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 9]
```diff
@@ -1,43 +1,73 @@
...
+
+By default, this value is false and the behaviour of the Archive Service is unchanged. However, if you are experiencing a `TransactionDeserializationException` or a `JsonMappingException` during the export of a JSON snapshot, this configuration option can be added to skip these transactions for a successful export. These transactions won’t be included in the export, but if a binary export is also created, all transactions can be preserved.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:58]:
```suggestion
By default, this value is false and the behaviour of the Archive Service is unchanged. However, if you are experiencing a `TransactionDeserializationException` or a `JsonMappingException` during the export of a JSON snapshot, this configuration option can be added to skip these transactions for a successful export. These transactions are not included in the export, but if a binary export is also created, all transactions can be preserved.
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 10]
```diff
@@ -1,43 +1,73 @@
...
+
+### Archive Service 1.0.2
+
+In this release:
+
+* Improved Archiving support of tokens when they are moved between more than one party and then redeemed.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:64]:
```suggestion
* The archiving support of tokens when they are moved between more than one party and then redeemed has been improved.
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 11]
```diff
@@ -1,43 +1,73 @@
...
+* Logging of the Archiving tool has been increased to aid in troubleshooting.
+* The Archiving client can now connect to nodes which are set up to use RPC SSL connection settings.
+* Tables are now ordered by table name length in descending order to prevent foreign key constraints from being violated when deleting rows.
+
+### Archive Service 1.0.1
+
+In this release:
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:71]:
```suggestion
```

File: content/en/tools/archiving-service/archiving-release-notes.md [hunk 12]
```diff
@@ -1,43 +1,73 @@
...
+
+* The Archive Service is compatible with [Ledger Graph V1.2.1 On Demand function]({{< relref "archiving-service-index.html#archiving-and-ondemand-ledgergraph" >}}).
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-release-notes.md:73]:
```suggestion
The Archive Service is now compatible with [Ledger Graph V1.2.1 On Demand function]({{< relref "archiving-service-index.html#archiving-and-ondemand-ledgergraph" >}}).
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 13]
```diff
@@ -1,419 +1,404 @@
...
+
+You can use Archive service commands to mark archivable items in your vault, archive them, and restore transactions from the archive when necessary.
+
+{{< note >}}
+Due to its in-memory design, the Archive Service is most effective and efficient for smaller ledgers & frequent archiving events. For scenarios that don't fit these constraints, you should consider building your required archiving logic within the application itself. This introduction includes advice on [making your CorDapps archive-friendly](#making-archive-friendly-cordapps).
+{{< /note >}}
+
+The Archive Service consists of the following:
+
+* [Archive Service CorDapp](#archive-service-cordapp)- which contains the necessary flows to mark and archive transactions.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:32]:
```suggestion
* [Archive Service CorDapp](#archive-service-cordapp) - contains the necessary flows to mark and archive transactions.
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 14]
```diff
@@ -1,419 +1,404 @@
...
+* [Archive Service Client Library](archive-library.md) - which provides programmatic access to the archive service, and [exposes relevant APIs](archiving-apis.md).
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:33]:
```suggestion
* [Archive Service Client Library](archive-library.md) - provides programmatic access to the archive service, and [exposes relevant APIs](archiving-apis.md).
* ```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 15]
```diff
@@ -1,419 +1,404 @@
...
+* [Archive Service Command Line Interface](archiving-cli.md) - which allows you to perform archiving tasks from the command line.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:34]:
```suggestion
* [Archive Service Command Line Interface](archiving-cli.md) - allows you to perform archiving tasks from the command line.
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 16]
```diff
@@ -1,419 +1,404 @@
...
+
+It also makes use of the [Application Entity Manager](app-entity-manager.md) - which allows CorDapps to access off-ledger databases using JPA APIs.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:36]:
```suggestion
It also makes use of the [Application Entity Manager](app-entity-manager.md), which allows CorDapps to access off-ledger databases using JPA APIs.
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 17]
```diff
@@ -1,419 +1,404 @@
...
+
+The Archive Service has commands you can use to identify which transactions can be archived in your vault. A fully consumed transaction or attachment will be marked as archivable when:
+
+* There are no unconsumed transactions in the same LedgerGraph component. A LedgerGraph component is a connected group of transactions, represented as a Direct Acrylic Graph (DAG) in the [LedgerGraph service]({{< relref "../ledgergraph/ledgergraph-index.md" >}}).
+* The transaction is not also referenced by another LedgerGraph component that contains unconsumed transactions.
+* The attachment itself is not a contract attachment.
+
+Archivable and non-archivable LedgerGraph components:
+
+![Archivable and non-archivable LG components](archivable-dags.png)
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:65]:
```suggestion

* ![Archivable and non-archivable LG components](archivable-dags.png)

```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 18]
```diff
@@ -1,419 +1,404 @@
...
+
+## When you can archive
+
+Once the Archive Service has marked a transaction or attachment as archivable, you can safely archive it anytime without risk to any other member of your network. You do not need to inform other members of the network, and your archiving action will not affect their ledger.
+
+### Archiving and Collaborative Recovery
+
+The [Collaborative Recovery CorDapp LedgerSync V1.2.1]({{< relref "../../platform/corda/4.9/enterprise/node/collaborative-recovery/ledger-sync.md" >}}) has been introduced for compatibility with the Archive Service. If you or anyone on your network uses Collaborative Recovery to recover data after a disaster scenario, archived transactions in any nodes on the network are marked as such. This means they do not appear to be 'lost' or 'missing' data and will not be recovered automatically in the recovery process.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:73]:
```suggestion
The [Collaborative Recovery CorDapp LedgerSync V1.2.1]({{< relref "../../platform/corda/4.9/enterprise/node/collaborative-recovery/ledger-sync.md" >}}) has been introduced for compatibility with the Archive Service. If you or anyone on your network uses Collaborative Recovery to recover data after a disaster scenario, archived transactions in any nodes on the network are marked as such. This means that they do not appear to be 'lost' or 'missing' data and will not be recovered automatically in the recovery process.
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 19]
```diff
@@ -1,419 +1,404 @@
...
+    url: "jdbc:postgresql:postgres"
+    user: "archive"
+    password: "<{HNtZpbrOGM6GhYA6foh5PCCBanUtaebCjauKL8ur9EE=:PolqmEJ7JOM+Sqj3ZNAE+Ew9bqG1wVE=}>"
+    schema: "archive"
+}
+```
+
+## Using the backup schema
+
+you can configure the archiving process to create a temporary snapshot image of the archivable transactions
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:163]:
```suggestion
You can configure the archiving process to create a temporary snapshot image of the archivable transactions
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 20]
```diff
@@ -1,419 +1,404 @@
...
+managed by different schema owners.
+
+The Archive Service uses a separate JPA entity manager factory to manage the
+archive schema and copy data from the Corda schema to the archive schema.
+
+To create a copy of the vault on the backup schema the backup schema owner must have
+`SELECT` rights to the vault schema.
+
+To restore the vault from a copy on the backup schema the backup schema owner must
+either have `INSERT` rights to the vault schema or the `restore-snapshot` command has to be executed with the
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:177]:
```suggestion
either have `INSERT` rights to the vault schema or the `restore-snapshot` command must be executed with the
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 21]
```diff
@@ -1,419 +1,404 @@
...
+create user archive password 'archive';
+create schema archive AUTHORIZATION archive;
+grant usage on schema corda to archive;
+grant select on all tables in schema corda to archive;
+grant insert on all tables in schema corda to archive;
+```
+
+### Oracle
+
+The following DDL statements can be used to create a backup schema user `archive`:
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:200]:
```suggestion
You can use the following DDL statements to create a backup schema user `archive`:
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 22]
```diff
@@ -1,419 +1,404 @@
...
+
+```
+create user archive identified by archive2 DEFAULT TABLESPACE users QUOTA unlimited ON users;
+grant CONNECT, RESOURCE to archive;
+```
+
+Oracle does not have a single grant option to provide rights to all the tables in a schema.
+This means that rights have to be current on a per table basis.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:208]: Not sure "current" is correct here?

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 23]
```diff
@@ -1,419 +1,404 @@
...
+grant CREATE VIEW to archive;
+grant SELECT on schema ::corda to archive;
+grant INSERT on schema ::corda to archive;
+```
+
+## Restarting the node
+
+It is sometimes necessary to restart the node when carrying out an archiving job.
+
+You need to restart the node:
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:232]:
```suggestion
You need to restart the node in the following circumstances:
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 24]
```diff
@@ -1,419 +1,404 @@
...
+```
+
+The property can be added to the Archive Service CorDapp configuration file, or passed within the
+`create-snapshot` command configuration.
+
+A suitable exporter, such as `QueryableStateFileExporter`, must also be listed on the command line to `export-snapshot`.
+
+## Additional tables
+
+Archive Service will automatically detect transaction and attachment tables which use the columns
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:385]:
```suggestion
Archive Service automatically detects transaction and attachment tables that use the columns
```

File: content/en/tools/archiving-service/archiving-service-index.md [hunk 25]
```diff
@@ -1,419 +1,404 @@
...
+`TRANSACTION_ID` or `ATT_ID` within the vault schema and include them in the archive process.
+
+Additional transaction and attachment tables which use different column names can be registered using the
+properties `additionalTransactionTables` and `additionalAttachmentTables` with the following format.
```
Comment (nadinequinr3) [inline content/en/tools/archiving-service/archiving-service-index.md:389]:
```suggestion
properties `additionalTransactionTables` and `additionalAttachmentTables` with the following format:
```

File: content/en/tools/archiving-service/archiving-service-index.md (hunk 22 above)
Comment (AnnaK-R3) [reply content/en/tools/archiving-service/archiving-service-index.md:211]: I didn't write this content so not sure either. Leaving as is.

**Pull request merged**